python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,491 @@
+            return f"{warehouse_path}/{database_name}.db/{table_name}"
+
+        raise ValueError("No default path is set, please specify a location when creating a table")
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str) -> str:
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    def _create_glue_table(self, identifier: Union[str, Identifier], table_input: Dict[str, Any]) -> None:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            self.glue.create_table(DatabaseName=database_name, TableInput=table_input)
+        except self.glue.exceptions.AlreadyExistsException as e:
+            raise TableAlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} not found") from e
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        """Create an Iceberg table in Glue catalog
+        Args:
+            identifier: Table identifier.
+            schema: Table's schema.
+            location: Location for the table. Optional Argument.
+            partition_spec: PartitionSpec for the table.
+            sort_order: SortOrder for the table.
+            properties: Table properties that can be a string based dictionary.
+        Returns:
+            Table: the created table instance
+        Raises:
+            AlreadyExistsError: If a table with the name already exists
+            ValueError: If the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        metadata_location = f"{location}/metadata/00000-{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order, properties=properties
+        )
+        io = load_file_io(properties=self.properties, location=metadata_location)
+        _write_metadata(metadata, io, metadata_location)
+
+        self._create_glue_table(
+            identifier=identifier, table_input=_construct_table_input(table_name, metadata_location, properties)
+        )
+        try:
+            loaded_table = self.load_table(identifier=(database_name, table_name))
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} fail to be created") from e
```

Comment:
Thank you for your suggestions.  I inspected the logic again and found that `load_table` would not throw any table/database related exception here since both `duplicated creation` and `database not exist` errors are caught and throws in `_create_glue_table` above. I think it may be ok to remove this `try...except...` statement here.